### DISCUSSION TITLE
How to use new_page

### DISCUSSION THREAD

**Discussion by nitinsinghal123** [dompdf/dompdf#2610] [General]
Hello

Can anyone please show me a working example of new_page()? This is my current code:

test.php:
```
   require_once 'dompdf/autoload.inc.php';
    use Dompdf\Dompdf;
    $dompdf = new Dompdf();
    $html = file_get_contents("test_html.php");
    $dompdf->loadHtml($html);
    $dompdf->setPaper('A4', 'potrait');
    $dompdf->render();
    ob_get_clean();
    $dompdf->stream("", array("Attachment" => false));
```

test_html.php:

````
<html>
	<body>
        <div> Page 1 </div>
        <div> Page 2 </div>
	</body>
</html>
````

I want both the div on different pages with the help of new_page() function. Any help would really be helpful.

Regards

**Comment by bsweeney** [comment]
You don't use the `new_page` method directly with Dompdf. It's an HTML+CSS parser so you would just indicate that you want to start a new page using CSS:

```
<html>
	<body>
        <div> Page 1 </div>
        <div style="page-break-before: always;"> Page 2 </div>
	</body>
</html>
```

**Comment by nitinsinghal123** [comment]
Thanks @bsweeney. It is working. 

Just for my knowledge, can you please tell me how do we use `new_page`? Maybe I can use that as well somewhere in my project.

Regards

**Comment by bsweeney** [comment]
You would just call it:

```
$dompdf->getCanvas()->new_page();
```

But it's not typically something you would do manually since Dompdf fully manages paging during HTML content rendering. If you're interested in manually manipulating the PDF you should use the underlying PDF generation library (CPDF or PDFLib).

**Comment by nitinsinghal123** [comment]
Got it. Thanks
